spec/services/filesync/filesync.spec.ts

```diff
@@ -58,11 +65,41 @@ describe("FileSync.init", () => {
     expect(filesync._syncJson).toEqual(state);
   });
 
+  it("loads state from an old .gadget/sync.json and ignores mtime", async () => {
+    const state = {
+      app: testApp.slug,
+      filesVersion: "77",
+      mtime: 1658153625236,
+    };
+
+    await fs.outputJSON(appDirPath(".gadget/sync.json"), state);
+
+    const filesync = await FileSync.init(makeContext({ parse: args, argv: ["sync", appDir, "--app", testApp.slug] }));
+
+    const { mtime: _, ...rest } = state; // mtime is removed,
+
+    // @ts-expect-error _syncJson is private
+    expect(filesync._syncJson).toEqual({
+      ...rest,
+      currentEnvironment: "development",
+      environments: {
+        development: { filesVersion: "77" },
+      },
+    });
+  });
```

Comment:
🙌 for making test this!

We've spoken IRL about this -- I'd love if we didn't delete the mtime so people could use an older version of `ggt` that requires it, but the more I think about, the more it's probably fine to just leave this as is 👍